lib/src/container/unencapsulate.rs

```diff
@@ -205,17 +205,38 @@ pub(crate) async fn fetch_layer_decompress<'a>(
     manifest: &oci_image::ImageManifest,
     layer: &'a oci_image::Descriptor,
     progress: Option<&'a Sender<Option<store::LayerProgress>>>,
+    layer_info: Option<&Vec<containers_image_proxy::ConvertedLayerInfo>>,
+    transport_src: Transport,
 ) -> Result<(
     Box<dyn AsyncBufRead + Send + Unpin>,
     impl Future<Output = Result<()>> + 'a,
 )> {
     use futures_util::future::Either;
     tracing::debug!("fetching {}", layer.digest());
     let layer_index = manifest.layers().iter().position(|x| x == layer).unwrap();
+    let (blob, driver, size);
+    let media_type: &oci_image::MediaType;
+    match transport_src {
+        Transport::ContainerStorage => {
+            let n_layers = layer_info.unwrap().len();
```

Comment:
Instead of `.unwrap()` here, let's do:

```suggestion
           let layer_info = layer_info.ok_or_else(|| anyhow!("skopeo is too old to pull from containers-storage"))?;
           let n_layers = layer_info.len();
```
etc.  (You'll also want to drop the `unwrap()` from the line below)

Reply:
Then what you can do for now is adjust the integration test to look for the string "skopeo is too old".

Once the skopeo PR lands and is built, we can change the integration test again to finally verify things work!